kotlinx-coroutines-core/native/src/Builders.kt

```diff
@@ -62,6 +63,18 @@ private class BlockingCoroutine<T>(
 ) : AbstractCoroutine<T>(parentContext, true, true) {
     private val joinWorker = Worker.current
 
+    /**
+     * Send a ping to the worker to prevent it from terminating while this coroutine is running,
+     * ensuring that continuations don't get dropped and forgotten.
+     */
+    fun keepAlive() {
+        Worker.current.executeAfter(afterMicroseconds = 100_000) {
+            if (!isCompleted) {
+                keepAlive()
```

Comment:
I'm not really fond of this change.
It has really weird semantics "let's invoke something on a regular basis to workaround the issue that only reproduces during termination", it interferes with nested `runBlocking`s (producing probably unnecessary tasks) and opens a can of worms for "what happens if 'start' or 'joinBlocking' throw system exception (e.g. OOM)". For now it seems it will create a task that is just stuck forever

Reply:
These are all good points, I fixed the code accordingly.

Regarding the weird semantics, I don't see anything weird about it. You could say, "to workaround the issue that only reproduces during termination" about any mechanism that recognizes something canceling or closing, yet such code is common. Here, we're just explicitly making sure the worker knows further work will be expected from it.

Reply:
Also, "keepalive" is a common term denoting exactly this approach, though from a different IT area: https://en.wikipedia.org/wiki/Keepalive